include/illixr/ProperQuaternion.hpp

```diff
@@ -0,0 +1,233 @@
+#pragma once
+#include <eigen3/Eigen/Dense>
+#include <iostream>
+
+namespace ILLIXR {
+
+/**
+ * @brief A more complete implementation of the Eigen::Quaternion
+ *
+ * The base class, Eigen::Quaternion, is missing some arithmetic functionality, like scalar multiplication,
+ * this implementation adds this functionality.
+ *
+ * @tparam Scalar_ The type of data the Quaternion holds (typically double ot float)
+ * @tparam Options_ Data alignment
+ */
+template<typename Scalar_, int Options_ = Eigen::AutoAlign>
+class ProperQuaternion : public Eigen::Quaternion<Scalar_, Options_> {
```

Comment:
Do we want to update the name of this struct according to our new naming policy?